eventuate-local-java-cdc-connector-mysql-binlog/src/test/java/io/eventuate/local/mysql/binlog/OffsetProcessorTest.java

```diff
@@ -0,0 +1,114 @@
+package io.eventuate.local.mysql.binlog;
+
+import io.eventuate.common.eventuate.local.BinlogFileOffset;
+import io.eventuate.local.db.log.common.OffsetStore;
+import io.eventuate.util.test.async.Eventually;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+public class OffsetProcessorTest {
```

Comment:
This is far too complicated.

First, use a mockito mock for OffsetStore.

Second, tests can be really simple and don't involve timeouts.
for example:

1. Create 3 CompletableFutures: a, b, c - new CompletableFuture()
2. Call `offsetProcessor.saveOffset()` with each one.
3. Complete them in reverse order: c.complete(..)
4. Verify that OffsetStore was called once with c's offset

Reply:
ok